src/Settings/BatteryIndicatorSettings.h

```diff
@@ -19,4 +19,21 @@ class BatteryIndicatorSettings : public SettingsGroup
     DEFINE_SETTING_NAME_GROUP()
 
     DEFINE_SETTINGFACT(display)
-};
+    DEFINE_SETTINGFACT(threshold1)
+    DEFINE_SETTINGFACT(threshold2)
+
+    Q_PROPERTY(bool visible READ visible NOTIFY visibleChanged)
```

Comment:
So all f this is incorrect:
* There is no need for adding a visible property. It already exists on both the group and on each setting. You can access them in Qml using `QGroundControl.settingsManager.batteryIndicatorSettings.visible` for visibility which should control whether to show the battery indicator panel. And then `QGroundControl.settingsManager.batteryIndicatorSettings.threshold1.visible` for the thresholds.
* To implement validation on the C++ side you need to implement you own settings fact creation using `DECLARE_SETTINGSFACT_NO_FUNC`. Look at AppSettings.cc for examples. Then in that code you connect to the rawValueChanged signal. And in you signal handler that is where you do the validation.

Reply:
I streamlined the visibility management by utilizing existing properties in QGroundControl.settingsManager.batteryIndicatorSettings and removed redundant properties. Additionally, I implemented validation for threshold settings using DECLARE_SETTINGSFACT_NO_FUNC 

![Screenshot from 2024-09-28 21-16-04](https://github.com/user-attachments/assets/cdc118fa-4293-460b-a639-963623b33415)

and confirmed the correct storage of new threshold values after verifying defaults from the JSON file.

![Screenshot from 2024-09-28 22-54-15](https://github.com/user-attachments/assets/8dc1e135-6d41-4cd7-a14f-49b64e6db874)

Thanks!